Source/WebCore/rendering/style/TextSizeAdjustment.h

```diff
@@ -28,18 +28,19 @@ namespace WebCore {
 
 class RenderStyle;
 
-enum TextSizeAdjustmentType { AutoTextSizeAdjustment = -1, NoTextSizeAdjustment = -2 };
-
 class TextSizeAdjustment {
 public:
-    constexpr TextSizeAdjustment() : m_value(AutoTextSizeAdjustment) { }
+    static constexpr float Auto = -1;
+    static constexpr float None = -2;
```

Comment:
To get some of the benefits of type safety we could use named constructors instead. Here’s an example of one way to do it that keeps things encapsulated.
```
public:
    static constexpr auto() { return TextSizeAdjustment(true); }
    static constexpr none() { return TextSizeAdjustment(false); }
    constexpr TextSizeAdjustment(float value) : m_value(value) { ASSERT_IN_CONSTEXPR_CONTEXT(value >= 0); }
private:
    static constexpr float Auto = -1;
    static constexpr float None = -2;
    constexpr TextSizeAdjustment(bool isAuto) : m_value(isAuto ? Auto : None) { }
```

Reply:
I don't think we can name the method `auto()` since `auto` is a C++ keyword.

Would using `Auto()` and `None()` violate code style guidelines?

Reply:
Yes it would. Could name it `autoAdjustment` if you like. Please double check though because I am not **sure** `auto` is disallowed.

Reply:
> Please double check though because I am not sure auto is disallowed.

It is not allowed:

```
Source/WebCore/rendering/style/TextSizeAdjustment.h:33:41: error: cannot combine with previous 'type-name' declaration specifier
    static constexpr TextSizeAdjustment auto() { return TextSizeAdjustment(true); }
                                        ^
Source/WebCore/rendering/style/TextSizeAdjustment.h:33:46: error: expected member name or ';' after declaration specifiers
    static constexpr TextSizeAdjustment auto() { return TextSizeAdjustment(true); }
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ ^
```

Reply:
Posted a PR using `autoAdjust()` and `none()` using your approach, Darin.

Also considered using `adjustAuto()` and `adjustNone()`, but I didn't want to over-think this after all the back-and-forth.